### PR TITLE
Switch to GitHub actions, pre-commit support, black

### PR DESCRIPTION
**Pull request by zerolab**
With Travis throttling OS projects, let's swithc to GitHub Actions.

While at it, I added pre-commit support and black, and update isort. As well as lint everything.

See it in action[sic!]: https://github.com/zerolab/wagtailmedia/pull/1

### CODE REVIEW

**Review by thibaudcolas — APPROVED** [review]
Thank you @zerolab, there’s some long overdue cleanup in there! A couple of questions and minor points – feel free to merge this as you see fit.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.experimental }}
+    strategy:
+      matrix:
+        include:
+          - python: 3.6
+            django: "Django>=2.2,<2.3"
+            wagtail: "wagtail>=2.10,<2.11"
+            experimental: false
+          - python: 3.7
```
[the remark this answers is not in the document]
Comment (zerolab) [reply .github/workflows/test.yml:24]: Tox has additional requirements to make it work with GitHub Actions. I went the Wagtail core way for GitHub Actions

File: setup.cfg [hunk 2]
```diff
@@ -6,12 +6,19 @@ description-file = README.md
 
 [flake8]
 max-line-length=120
-exclude=migrations
+exclude=migrations,dist,build,vagrant,venv
```
[the remark this answers is not in the document]
Comment (zerolab) [reply setup.cfg:9]: I've added `extend-ignore = E203`, but left max-line-length at 120 as we do the same on other projects including WK. If we need to bring it down to 88, that can be done later

**Comment by zerolab** [comment]
Tweaked a bit, rebased and pushed. https://github.com/zerolab/wagtailmedia/pull/1 is green, so merging without waiting for Travis. Thank you @thibaudcolas

**Pull request merged**